sources/appendix-value-extraction.asciidoc

```diff
@@ -0,0 +1,437 @@
+TODO: format of key representations
+* `java.util.Optional`. No node will be appended to the path when validating type-level constraints on `Optional`. Constraints given on an element of type `Optional` apply to the wrapped value by default (see below).
+* `javafx.beans.observable.ObservableValue`. No node will be appended to the path when validating type-level constraints on `ObservableValue`. Constraints given on an element of type `ObservableValue` apply to the wrapped value by default (see below).
+
+==== Plugging in custom extractors
+
+Additional value extractors can be registered with the Bean Validation provider in the following ways:
+
+* Invoke the new method `Configuration#addValueExtractor(ValueExtractor<?>)` (to apply it on the validation factory level)
+* Invoke the new method `ValidatorContext#addValueExtractor(ValueExtractor<?>)` (to apply it for a single `Validator`)
+* Specify the fully qualified class name of one or several extractors in `META-INF/validation.xml`: 
+
+[source,xml]
+----
+<value-extractors>
+    <value-extractor>com.example.MyExtractor</value-extractor>
+</value-extractors>
+----
+
+==== Retrieval of extractors
+
+When detecting a type-level constraint or cascade, the applicable extractor is determined as follows:
+
+* Choose the most specific extractor matching the type parameter in question; an extractor A is more specific than another extractor B if A extracts a subtype of the type extracted by B.
+* If there are several extractors which are equally specific, an `UnexpectedTypeException` is raised. TODO: apply rules similar to "ConstraintValidator resolution algorithm".
+* When obtaining the extractor for type parameter constraint validation, the declared type of the validated element is used. This is to be consistent with constraint validator resolution, which is based on the static type of elements, not the runtime type.
+* When obtaining the extractor for cascaded validation, the runtime type of the cascasded element is used. This is to be consistent with the rules defined for property path construction which are based on the runtime type.
+
+[NOTE]
+.Implementation note
+====
+As extractor retrieval for type parameter constraints is done using the static type of constrained elements, the retrieval can be done once at initialization time and then be cached. This is not possible for retrieval of extractors for cascaded validation.
+====
```

Comment:
I think you answered the question by no. You do not want the mechanism to extract values that are not related to a type parameter.

I'm only slightly uneased by it as it might block us in a future where we want to use the extractor for more things that type parameter validation and `@Valid`. I don't have concrete examples of such future however.